dask_image/ndinterp/__init__.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+
+
+import numpy as np
+import dask.array as da
+from scipy import ndimage
+
+
+__all__ = [
+    "affine_transform",
+]
+
+
+def affine_transform(
+        input,
+        matrix,
+        offset=0.0,
+        output_shape=None,
+        output_chunks=None,
+        **kwargs
+):
+    """Apply an affine transform using Dask.
+    `ndimage.affine_transformation` for Dask Arrays. For every
+    output chunk, only the slice containing the relevant part
+    of the input is passed on to `ndimage.affine_transformation`.
+
+    Notes
+    -----
+        Differences to `ndimage.affine_transformation`:
+        - currently, prefiltering is not supported
+          (affecting the output in case of interpolation `order > 1`)
+        - default order is 1
+        - currently, matrix is expected to have shape (ndim,) or (ndim, ndim)
+          (because offset needs to be set for each chunk)
+        - modes 'reflect', 'mirror' and 'wrap' may yield different results
+
+        To do:
+        - optionally use cupyx.scipy.ndimage.affine_transform
+
+        Arguments equal to `ndimage.affine_transformation`,
+        except for `output_chunks`.
+
+    Parameters
+    ----------
+    input : array_like (Numpy Array, Dask Array, ...)
+        The input array.
+    matrix : array (ndim,) or (ndim, ndim)
+        Transformation matrix.
+    offset : array (ndim,)
+        Transformation offset.
+    output_shape : array (ndim,), optional
+        The size of the array to be returned.
+    output_chunks : array (ndim,), optional
+        The chunks of the output Dask Array.
+
+    Returns
+    -------
+    affine_transform : Dask Array
+        A dask array representing the transformed output
+
+    """
+
+    if output_shape is None:
+        output_shape = input.shape
+
+    # process kwargs
+
+    # set default order to 1, warn if different
+    if 'order' in kwargs:
+        order = kwargs['order']
+        del kwargs['order']
+        if order > 1:
+            UserWarning('Currently, for order > 1 the output of'
```

Comment:
This is a very vague user warning, what actually happens here if order > 1 and how is it different that what we might expect?

Reply:
You're right, added: "Specifically, no spline prefiltering is performed and therefore the output can contain more blur."